specification/alertsmanagement/resource-manager/Microsoft.AlertsManagement/preview/2023-01-01-preview/AlertRuleRecommendations.json

```diff
@@ -0,0 +1,251 @@
+            "schema": {
+              "$ref": "#/definitions/AlertRuleRecommendationsListResponse"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "List alert rule recommendations for virtual machines at resource level": {
+            "$ref": "./examples/AlertRuleRecommendations_GetByResource_VM.json"
+          },
+          "List alert rule recommendations for Monitoring accounts at resource level": {
+            "$ref": "./examples/AlertRuleRecommendations_GetByResource_MAC.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.AlertsManagement/alertRuleRecommendations": {
+      "get": {
+        "tags": [
+          "alertRuleRecommendations"
+        ],
+        "description": "Retrieve alert rule recommendations for a target type.",
+        "operationId": "AlertRuleRecommendations_ListByTargetType",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/TargetTypeParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Successful request for a list of alert rule recommendations",
+            "schema": {
+              "$ref": "#/definitions/AlertRuleRecommendationsListResponse"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "List alert rule recommendations for virtual machines at subscription level": {
+            "$ref": "./examples/AlertRuleRecommendations_GetBySubscription_VM.json"
+          },
+          "List alert rule recommendations for Monitoring accounts at subscription level": {
+            "$ref": "./examples/AlertRuleRecommendations_GetBySubscription_MAC.json"
+          }
+        }
+      }
+    }
+  },
+  "parameters": {
+    "ResourceUriParameter": {
+      "name": "resourceUri",
+      "in": "path",
+      "required": true,
+      "type": "string",
+      "description": "The identifier of the resource.",
+      "x-ms-parameter-location": "method",
+      "x-ms-skip-url-encoding": true
+    },
+    "TargetTypeParameter": {
```

Comment:
>TargetTypeParameter [](http://example.com/codeflow?start=5&length=19)

Can this be an enum? Like is there a pre-defined list of target types?

Reply:
Not in this case - the value can be any resource type or a target scenario like 'AzureBackup'.